ui/app/store/actions.js

```diff
@@ -2660,6 +2662,23 @@ function setUseNonceField (val) {
   }
 }
 
+function setIpfsGateway (val) {
+  return (dispatch) => {
+    dispatch(actions.showLoadingIndication())
+    log.debug(`background.setIpfsGateway`)
+    background.setIpfsGateway(val, (err) => {
+      dispatch(actions.hideLoadingIndication())
+      if (err) {
+        return dispatch(actions.displayWarning(err.message))
+      }
+    })
+    dispatch({
```

Comment:
Should this be inside the callback above?

Reply:
I put it inside the callback, only calling it if there's no error.